api/v1beta3/cloudstackcluster_types.go

```diff
@@ -34,6 +34,10 @@ type CloudStackClusterSpec struct {
 
 	// The kubernetes control plane endpoint.
 	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// SyncWithACS determines if an externalManaged CKS cluster should be created on ACS.
+	// +optional
+	SyncWithACS bool `json:"syncWithACS,omitempty"`
```

Comment:
nit: I could be wrong bu I thought the recommended way was to use pointers for optional fields? I think that's what I remember from the k8s api conventions

Reply:
As per my understanding we don't need to do that here. Pointer is used if we want to distinguish between explicit `false` and not specified.

Ref.: https://book.kubebuilder.io/cronjob-tutorial/api-design.html - search for `pointer` on the page.

Reply:
I was talking about this: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Again not required, but k8s does seem to recommend using pointer if the type doesnt have a built-in nil.

On the topic of needing to distinguish unset vs false:

> Using a pointer allows distinguishing unset from the zero value for that type. There are some cases where, in principle, a pointer is not needed for an optional field since the zero value is forbidden, and thus implies unset. There are examples of this in the codebase. However:
>
>*    it can be difficult for implementors to anticipate all cases where an empty value might need to be distinguished from a zero value
>*   structs are not omitted from encoder output even where omitempty is specified, which is messy;
>*   having a pointer consistently imply optional is clearer for users of the Go language client, and any other clients that use corresponding types

Reply:
@vishesh92 are you planning on changing this? or what's the decision here?
Otherwise the PR looks good to go

Reply:
I was thinking of keeping it as bool only.

Reply:
@g-gaston I have updated the PR to make it a pointer to a bool. Please check now.